src/plugins/GUITestBase/src/tests/common_scenarios/primer3/GTTestsCommonScenariosPrimer3.cpp

```diff
@@ -520,6 +522,41 @@ GUI_TEST_CLASS_DEFINITION(test_0023) {
 
 }
 
+GUI_TEST_CLASS_DEFINITION(test_0024) {
+    GTFileDialog::openFile(os, testDir + "_common_data/primer3", "human.fa");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    Primer3DialogFiller::Primer3Settings settings;
+    settings.filePath = testDir + "_common_data/primer3/input/test_0024.txt";
+    settings.loadManually = false;
+    settings.notRun = true;
+    GTUtilsDialog::add(os, new Primer3DialogFiller(os, settings));
+    GTToolbar::clickButtonByTooltipOnToolbar(os, MWTOOLBAR_ACTIVEMDI, "Primer3");
+
+    class Scenario : public Filler {
+    public:
+        Scenario(HI::GUITestOpStatus& _os)
+            : Filler(_os, "Primer3Dialog") {
+        }
+        void run() override {
+            QWidget* dialog = GTWidget::getActiveModalWidget(os);
+            GTUtilsDialog::add(os, new GTFileDialogUtils(os, sandBoxDir, "test_0024.txt", GTFileDialogUtils::Save));
+            GTWidget::click(os, GTWidget::findWidget(os, "saveSettingsButton", dialog));
+
+            if (isOsMac()) {
+                dialog->close();
```

Comment:
Same here: mimic a real user behavior in UI tests. A user can't call API.

Reply:
OK